src/ir.cc

```diff
@@ -379,32 +374,31 @@ ScriptModule::ScriptModule(Type type) : type(type) {
     case ScriptModule::Type::Binary:
       new (&binary.loc) Location();
       new (&binary.name) std::string();
-      binary.data = nullptr;
-      binary.size = 0;
+      new (&binary.data) std::vector<uint8_t>();
       break;
 
     case ScriptModule::Type::Quoted:
       new (&quoted.loc) Location();
       new (&quoted.name) std::string();
-      quoted.data = nullptr;
-      quoted.size = 0;
+      new (&quoted.data) std::vector<uint8_t>();
       break;
   }
 }
 
 ScriptModule::~ScriptModule() {
   typedef std::string std_string;
+
   switch (type) {
     case ScriptModule::Type::Text:
       delete text;
       break;
     case ScriptModule::Type::Binary:
       binary.name.~std_string();
-      delete [] binary.data;
+      binary.data.~vector();
```

Comment:
Yes. It is the placement call to the destructor, to match the placement call to the constructor in the previous method. Placement construction/destruction is necessary for unions.